walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,135 @@
+    
+    List<TimeRange> mandatoryAttendeeAvailability = new ArrayList<>();
+    List<TimeRange> allAttendeeAvailability = new ArrayList<>();
+
+    //if the meeting is scheduled for longer than a day, it cannot be scheduled
+    if (request.getDuration() > 24*60) {
+      return allAttendeeAvailability;
+    } 
+
+    //if no attendees or no meetings, any time of day works
+    else if ((request.getAttendees().size() == 0 && request.getOptionalAttendees().size() == 0) || events.size() == 0) {
+      allAttendeeAvailability.add(TimeRange.WHOLE_DAY);
+      return allAttendeeAvailability;
+    } 
+
+    //otherwise, find all possible meeting times   
+    else {
+
+      //first find the availability for mandatory attendees only 
+      List<TimeRange> mandatoryAttendeeMeetingTimes = 
+          findAndSortConflictingMeetings(events, request, false);
+      
+      fixNestedOrOverlappingMeetings(mandatoryAttendeeMeetingTimes);
+      mandatoryAttendeeAvailability = findAvailabilityBetweenMeetings(mandatoryAttendeeMeetingTimes, request);
+
+      //if optional attendees, find availability for all attendees and compare to availability for mandatory attendees 
+      if (request.getOptionalAttendees().size() > 0) {
+        List<TimeRange> allAttendeeMeetingTimes = 
+            findAndSortConflictingMeetings(events, request, true);
+
+        fixNestedOrOverlappingMeetings(allAttendeeMeetingTimes);
+        allAttendeeAvailability = findAvailabilityBetweenMeetings(allAttendeeMeetingTimes, request);
+
+        //compare mandatory attendees' availability with optional attendees' availability
+        if (allAttendeeAvailability.size() > 0 || request.getAttendees().size() == 0) {
+          return allAttendeeAvailability;
+        }
+      }
+    }
+    return mandatoryAttendeeAvailability;
+  }
+
+  //only keep meetings that have overlapping attendees, and sort by start time 
+  public List<TimeRange> findAndSortConflictingMeetings(Collection<Event> events, MeetingRequest request, Boolean allAttendees) {
+    List<TimeRange> conflictingMeetings = new ArrayList<>();
+    
+    for (Iterator iterator = events.iterator(); iterator.hasNext();) {
+      List<String> newMeetingAttendees = new ArrayList<>();
+      if (allAttendees) {
+        newMeetingAttendees.addAll(0, request.getOptionalAttendees());
+      }
+      newMeetingAttendees.addAll(0, request.getAttendees());
+
+      Event event = (Event) iterator.next();
+      List<String> existingMeetingAttendees = new ArrayList<>(event.getAttendees());
+      newMeetingAttendees.retainAll(existingMeetingAttendees); 
+      
+      if (newMeetingAttendees.size() != 0) {
+        conflictingMeetings.add(event.getWhen());
+      }
+    }
+    
+    Collections.sort(conflictingMeetings, Comparator.comparingInt(TimeRange::start)); 
+    return conflictingMeetings;
+  }
+
+  //remove nested and merge overlapping time ranges
+  public void fixNestedOrOverlappingMeetings(List<TimeRange> meetingTimes) {
```

Comment:
This might be more succinctly named `mergeOverlappingMeetings`, since that's exactly what it does. It makes it a bit less ambiguous since "fixing" those meetings could mean different things.

Reply:
Thanks, I was trying to think of a better name because I new it was a little verbose and unclear, but hadn't changed it. And as for all of my styling errors, I was planning on going back to check the guide earlier, but then had a meeting. I should have not submitted the pull request in the first place until I had checked on the styling, so I apologize, I just meant to say that I recognized that I messed up some things. I appreciate the feedback!